kernel/src/utilities/capability_ptr.rs

```diff
@@ -56,7 +50,7 @@ impl From<CapabilityPtr> for usize {
     /// Provenance note: may not expose provenance.
     #[inline]
     fn from(from: CapabilityPtr) -> Self {
-        from.ptr as usize
+        from.ptr.into()
```

Comment:
`From<CapabilityPtr> for usize` is lossy and should be replaced with an inherent method. I suggest `addr` to copy [pointer's inherent addr method](https://doc.rust-lang.org/std/primitive.pointer.html#method.addr).

Reply:
Moved to an inherent `addr` method in #4332 .